x/restaking_assets_manage/types/keys.go

```diff
@@ -81,40 +88,64 @@ var (
 	KeyPrefixReStakerAssetInfos = []byte{prefixRestakerAssetInfo}
 
-	// KeyPrefixOperatorAssetInfos key->value: operatorAddr+'_'+AssetId->OperatorSingleAssetInfo
-	// or operatorAddr->mapping(AssetId->OperatorSingleAssetInfo) ?
+	// KeyPrefixOperatorAssetInfos key->value: operatorAddr+'_'+AssetID->OperatorSingleAssetInfo
+	// or operatorAddr->mapping(AssetID->OperatorSingleAssetInfo) ?
 	KeyPrefixOperatorAssetInfos = []byte{prefixOperatorAssetInfo}
 
-	// KeyPrefixOperatorOptedInMiddleWareAssetInfos key->value: operatorAddr+'_'+AssetId->mapping(middleWareAddr->struct{})
-	// or operatorAddr->mapping(AssetId->mapping(middleWareAddr->struct{})) ?
-	KeyPrefixOperatorOptedInMiddleWareAssetInfos = []byte{prefixOperatorOptedInMiddlewareAssetInfo}
+	// KeyPrefixOperatorOptedInMiddleWareAssetInfos key->value:
+	// operatorAddr+'_'+AssetID->mapping(middleWareAddr->struct{})
+	// or operatorAddr->mapping(AssetID->mapping(middleWareAddr->struct{})) ?
+	KeyPrefixOperatorOptedInMiddleWareAssetInfos = []byte{
+		prefixOperatorOptedInMiddlewareAssetInfo,
+	}
 
-	// KeyPrefixReStakerExoCoreAddr reStakerId = clientChainAddr+'_'+ExoCoreChainIndex
-	// KeyPrefixReStakerExoCoreAddr key-value: reStakerId->exoCoreAddr
+	// KeyPrefixReStakerExoCoreAddr reStakerID = clientChainAddr+'_'+ExoCoreChainIndex
+	// KeyPrefixReStakerExoCoreAddr key-value: reStakerID->exoCoreAddr
 	KeyPrefixReStakerExoCoreAddr = []byte{prefixRestakerExocoreAddr}
-	// KeyPrefixReStakerExoCoreAddrReverse k->v: exocoreAddress -> map[clientChainIndex]clientChainAddress
+	// KeyPrefixReStakerExoCoreAddrReverse k->v: exocoreAddress ->
+	// map[clientChainIndex]clientChainAddress
 	// used to retrieve all user assets based on their exoCore address
 	KeyPrefixReStakerExoCoreAddrReverse = []byte{prefixRestakerExocoreAddrReverse}
 )
 
-// GetAssetStateKey assetStateKey = stakerID+'/'+assetID
-func GetAssetStateKey(stakerID, assetID string) []byte {
-	return []byte(strings.Join([]string{stakerID, assetID}, "/"))
+func GetJoinedStoreKey(keys ...string) []byte {
+	return []byte(strings.Join(keys, "/"))
 }
 
-func ParseStakerAndAssetIDFromKey(key []byte) (stakerID string, assetID string, err error) {
+func ParseJoinedKey(key []byte) (keys []string, err error) {
 	stringList := strings.Split(string(key), "/")
-	if len(stringList) != 2 {
-		return "", "", errorsmod.Wrap(ErrParseAssetsStateKey, fmt.Sprintf("the stringList is:%v", stringList))
+	return stringList, nil
+}
+
+func ParseJoinedStoreKey(key []byte, number int) (keys []string, err error) {
+	stringList := strings.Split(string(key), "/")
+	if len(stringList) != number {
+		return nil, errorsmod.Wrap(ErrParseAssetsStateKey, fmt.Sprintf("expected length:%d,actual length:%d,the stringList is:%v", number, len(stringList), stringList))
 	}
-	return stringList[0], stringList[1], nil
+	return stringList, nil
+}
+
+// add for dogfood
+func OperatorSnapshotKey(operatorAddr sdk.AccAddress, height uint64) []byte {
```

Comment:
These snapshot functions can be removed.

Reply:
Okay